src/bpftrace.cpp

```diff
@@ -2196,6 +2196,37 @@ std::string BPFtrace::get_string_literal(const ast::Expression *expr) const
   return "";
 }
 
+std::optional<long> BPFtrace::get_int_literal(const ast::Expression *expr) const
+{
+  if (expr->is_literal)
+  {
+    if (auto *integer = dynamic_cast<const ast::Integer *>(expr))
+      return integer->n;
+    else if (auto *pos_param = dynamic_cast<const ast::PositionalParameter *>(
+                 expr))
+    {
+      if (pos_param->ptype == PositionalParameterType::positional)
+      {
+        auto param_str = get_param(pos_param->n, false);
+        if (is_numeric(param_str))
+          return std::stol(param_str);
+        else
+        {
+          LOG(ERROR, pos_param->loc)
+              << "$" << pos_param->n << " used numerically but given \""
+              << param_str << "\"";
+          return std::nullopt;
+        }
+      }
+      else
+        return (long)num_params();
+    }
+  }
+
+  LOG(ERROR) << "Expected integer literal, got " << expr->type;
```

Comment:
I think we should drop this error, i see every user of this already has its own error handling. Adding an extra error without location on top of that doesn't sound helpful. 

Or we could return the error too and append that to the error message, that would be useful but somewhat tricky to do. Std::variant works somewhat ok

Reply:
I agree with dropping the error, it doesn't really say much.

Returning the error and appending it to the message could be useful for the other error in this function (line 2215) as it always prints to the default `stderr` even if the function is called from semantic analyser (which normally prints to its own `err_` stream). The question is whether it is worth the effort.